extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -28,75 +34,136 @@ public static function onBeforePageDisplay() {
 		return true;
 	}
 
-	public static function render( $input, array $args ) {
-		global $wgCityId;
+	/**
+	 * Checks arguments for errors.
+	 * @param array $args
+	 * @param string errorMessage Return parameter with the proper error message to show. Disregard if return is false
+	 * @return true if ok, false if error
+	 */
+	private function checkArguments( array $args, $modelData, &$errorMessage ) {
+		// mostrecent must be bool
+		if ( isset( $args['mostrecent'] ) &&
+			$args['mostrecent'] !== 'true' &&
+			$args['mostrecent'] !== 'false'
+		) {
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
+				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+
+			return false;
+		}
 
-		$showLatest = !empty( $args['mostrecent'] ) && filter_var( $args['mostrecent'], FILTER_VALIDATE_BOOLEAN );
-		$itemCount = empty( $args['size'] ) ? self::ITEMS_DEFAULT : intval( $args['size'] );
-		$columns = empty( $args['columns'] ) ? self::COLUMNS_DEFAULT : intval( $args['columns'] );
-		$category = empty( $args['category'] ) ? '' :  $args['category'];
+		// size must be integer in range
+		if ( isset( $args['size'] ) ) {
+			$size = $args['size'];
 
-		if ( $itemCount > self::ITEMS_MAX ) {
-			$itemCount = self::ITEMS_MAX;
-		}
+			// TODO: Refactor error messages to avoid direct concatenation, see JPN-657
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'size' )->plain() .
```

Comment:
Do these error messages need to be set if we don't return false? i.e. should this be within the if block below?